src/bare-metal/alloc-example/src/main.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: Alloc
+#![no_main]
+#![no_std]
+
+extern crate alloc;
+extern crate panic_halt as _;
```

Comment:
Same question, any reason do this import?

Reply:
It's to provide a panic handler, rather than writing one ourselves and making the example longer.

Reply:
> It's to provide a panic handler, rather than writing one ourselves and making the example longer.

Right, and what you're implicitly saying is that even a "no-op import" like this has an effect? I'm used to only using imports to bring symbols into scope — but here the import also does something more behind the scenes?

Reply:
Ah, yes, the `extern crate` is necessary for the crate to be linked in, as we are not using any items from it in the code. If we leave this line out then it will fail to compile due to a missing panic handler. I've added a mention of this to the speaker notes.